pkg/infra/filestorage/db_filestorage.go

```diff
@@ -2,71 +2,103 @@ package filestorage
 
 import (
 	"context"
+	"crypto/md5"
+	"encoding/hex"
+
+	// can ignore because we don't need a cryptographically secure hash function
+	// sha1 low chance of collisions and better performance than sha256
+	// nolint:gosec
+	"crypto/sha1"
+	"fmt"
 	"strings"
 	"time"
 
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
 	"github.com/grafana/grafana/pkg/util/errutil"
 )
 
 type file struct {
-	Path             string    `xorm:"path"`
-	ParentFolderPath string    `xorm:"parent_folder_path"`
-	Contents         []byte    `xorm:"contents"`
-	Updated          time.Time `xorm:"updated"`
-	Created          time.Time `xorm:"created"`
-	Size             int64     `xorm:"size"`
-	MimeType         string    `xorm:"mime_type"`
+	Path                 string    `xorm:"path"`
+	PathHash             string    `xorm:"path_hash"`
+	ParentFolderPathHash string    `xorm:"parent_folder_path_hash"`
+	Contents             []byte    `xorm:"contents"`
+	ETag                 string    `xorm:"etag"`
+	CacheControl         string    `xorm:"cache_control"`
+	ContentDisposition   string    `xorm:"content_disposition"`
+	Updated              time.Time `xorm:"updated"`
+	Created              time.Time `xorm:"created"`
+	Size                 int64     `xorm:"size"`
+	MimeType             string    `xorm:"mime_type"`
 }
 
 type fileMeta struct {
-	Path  string `xorm:"path"`
-	Key   string `xorm:"key"`
-	Value string `xorm:"value"`
+	PathHash string `xorm:"path_hash"`
+	Key      string `xorm:"key"`
+	Value    string `xorm:"value"`
 }
 
 type dbFileStorage struct {
 	db  *sqlstore.SQLStore
 	log log.Logger
 }
 
+func createPathHash(path string) (string, error) {
+	hasher := sha1.New()
+	if _, err := hasher.Write([]byte(strings.ToLower(path))); err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
+}
+
+func createContentsHash(contents []byte) string {
+	hash := md5.Sum(contents)
+	return hex.EncodeToString(hash[:])
+}
+
 func NewDbStorage(log log.Logger, db *sqlstore.SQLStore, filter PathFilter, rootFolder string) FileStorage {
 	return newWrapper(log, &dbFileStorage{
 		log: log,
 		db:  db,
 	}, filter, rootFolder)
 }
 
-func (s dbFileStorage) getProperties(sess *sqlstore.DBSession, lowerCasePaths []string) (map[string]map[string]string, error) {
+func (s dbFileStorage) getProperties(sess *sqlstore.DBSession, pathHashes []string) (map[string]map[string]string, error) {
 	attributesByPath := make(map[string]map[string]string)
 
 	entities := make([]*fileMeta, 0)
-	if err := sess.Table("file_meta").In("path", lowerCasePaths).Find(&entities); err != nil {
+	if err := sess.Table("file_meta").In("path_hash", pathHashes).Find(&entities); err != nil {
```

Comment:
I wonder why? My stance on ORMs is that they are _great_ for simple SQL statements (such as the one above) as they save us from writing a lot of repetitive code, but should be avoided for more complicated stuff which requires in-depth knowledge of the particular ORM library.

Reply:
There are simple SQL statements (such as `SELECT … FROM table WHERE x = y`) where the ORM doesn’t end up doing what it should be doing in some cases because we’re neither strict about using pointers for every `x` nor about ensuring that `x` is not set to its zero value.

This has, in practice, already led to bugs (including vulns) and while there are several routes to go to fix it, using an ORM forces us to have a deep understanding of the ORMs behavior and quirks — and to spread that understanding to all the service authors to avoid them writing a bad query (which is easier to do than to not do). Using a query builder (such as http://doug-martin.github.io/goqu/ – though I haven’t evaluated this or any one for actual adoption) could allow us to keep the best part of the ORM (working with SQL in the native, compiled, language) without the quirks that comes from the abstraction introduced by ORMs.

SQL still is a footgun as far as languages go, anyone who’ve forgotten a WHERE in an UPDATE knows that, but when the abstraction layers doesn’t solve _that_, the question of what they’re supposed to solve remains.

NB: there is no proposal or design document that expands on this yet. That’ll come in time.

Reply:
@sakjur thanks! So, if I got this right, we don't want to use neither ORM nor raw SQL queries, but we currently do not have an alternative. Shouldn't we opt for the terser/more concise option while we are still doing the research, as it's going to be refactored anyway?

Reply:
I'd agree with that, it's essentially up to the service owners in lieu of more research, @papagian might disagree with me here though 🙂

Reply:
@sakjur @papagian wdyt about discussing it internally and adding the conclusion (whatever that is - ORM ban, sql ban, FFA) to the documentation, similar to https://github.com/grafana/grafana/pull/48663? I don't have super strong opinions on this, but I think its useful to have consistent rules

Reply:
Yes, but I'd much rather document the pros/cons of one choice versus the other than dictate a preferred common direction as long as we have two bad choices to pick between. I'm not too fond of forced consistency, I believe it, at its extreme, fosters a culture of design by compliance to rules rather than thoughtful design.

We should nudge people right and provide the documentation, tools, and examples necessary to lead by example and the autonomy to make informed and correct decisions for their needs.

Reply:
@sakjur I definitely agree, "discussing it internally" was not the best way to phrase it. Having an inclusive discussion, rather than a top down mandate, is definitely the way to go :) 

I mostly wanted to say that we need someone to drive the initiative _in this specific case_, as otherwise we will keep repeating the same discussion. Note - driving the initiative does not necessarily mean doing it all by oneself - getting others to participate is part of that job 😁